CondFormats/PPSObjects/interface/PPSAssociationCuts.h

```diff
@@ -70,7 +69,13 @@ class PPSAssociationCuts {
 
   ~PPSAssociationCuts() {}
 
-  const CutsPerArm &getAssociationCuts(const int sector) const { return association_cuts_.at(sector); }
+  // builds run-time data members, useful e.g. after loading data from DB
+  void initialize() {
+    for (auto &p : association_cuts_)
+      p.second.buildFunctions();
+  }
+
+  const CutsPerArm &getAssociationCuts(const int sector) const { return association_cuts_.find(sector)->second; }
```

Comment:
I just want to point out that with this approach a incorrect payload or `sector` value lead to undefined behavior (crashes at best, silent garbage output at worst).

Reply:
doing a check and throw (ie abort) at inizialize time is ok.
What is NOT ok is to check validity at each single access